core/unit_test/CMakeLists.txt

```diff
@@ -787,7 +787,7 @@ if(Kokkos_ENABLE_CUDA)
   KOKKOS_ADD_EXECUTABLE_AND_TEST(
     CoreUnitTest_CudaInterOpStreamsMultiGPU
     SOURCES
-      UnitTestMain.cpp
+      UnitTestMainInit.cpp
```

Comment:
What is this change for?

Reply:
This ensures that `Kokkos` is initialized before any tests are run and finalized after. The file used previously leaves that to the individual tests (which means that only one test case can be used since `Kokkos` can only be initialized and finalized once).